File: Protogen/loxim_protocol/java/src/main/java/pl/edu/mimuw/loxim/protocol/packages_data/LinkPackage.java
package pl.edu.mimuw.loxim.protocol.packages_data;

import java.util.EnumSet;
import pl.edu.mimuw.loxim.protogen.lang.java.template.ptools.Package;
import pl.edu.mimuw.loxim.protogen.lang.java.template.exception.ProtocolException;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageInputStreamReader;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageOutputStreamWriter;


import pl.edu.mimuw.loxim.protocol.enums.*;


public  class LinkPackage extends Package {
		public final static long ID=(long)129;
	
    	/**
**/
    	private Long valueId=new Long(0);
    	
	public LinkPackage()
	{
	}

		public LinkPackage(
		    	Long a_valueId    	)
	{
        	valueId=a_valueId;
    
	}
		
		@Override
	public long getPackageType() {
		return (long)ID;
	}
		
	@Override
	protected void deserializeW(PackageInputStreamReader reader)
			throws ProtocolException {
		super.deserializeW(reader);
		
				try{
	    			valueId=reader.readVaruint();
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}
			}
	
	@Override
	protected void serializeW(PackageOutputStreamWriter writer)
			throws ProtocolException {
		super.serializeW(writer);
				try{
	    			writer.writeVaruint(valueId);		
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}	
			}
	
	@Override
	public boolean equals(Object obj) {
		if(!super.equals(obj))
			return false;
		LinkPackage p=(LinkPackage)obj;
	    if (!(((valueId==p.getValueId())||((valueId!=null)&&(p.getValueId()!=null)&&((valueId).equals(p.getValueId())))))) 
return false;
				return true;
	}
	
//================= GETTERS AND SETTERS ===================

        	public Long getValueId(){return valueId;};
		
    	public void setValueId(Long a_valueId){ valueId=a_valueId;};
		
    
	 
}

File: Protogen/loxim_protocol/cpp/protocol/packages_data/LinkPackage.cpp
#include <string.h>
#include <stdlib.h>
#include "LinkPackage.h"
#include "ptools/PackageBufferWriter.h"
#include "ptools/PackageBufferReader.h"

using namespace protocol;

LinkPackage::LinkPackage(\
		int64_t a_valueId
		)
{
	valueId=a_valueId;
}

LinkPackage::LinkPackage()
{
	valueId=0;
}

LinkPackage::~LinkPackage()
{
}

void LinkPackage::serializeW(PackageBufferWriter *writer)
{
	Package::serializeW(writer);
	writer->writeVaruint(valueId);
}

bool LinkPackage::deserializeR(PackageBufferReader *reader)
{
	if (!Package::deserializeR(reader)) return false;
	{bool b; valueId=reader->readVaruint(b);}
	return true;
}

bool LinkPackage::equals(Package *_package)
{
	if (!_package) return false;
	if (_package->getPackageType()!=this->getPackageType()) return false;
	if (!Package::equals(_package)) return false;
	LinkPackage* p=(LinkPackage*)_package;
	if (!(getValueId()==p->getValueId())) return false;
	return true;
}
